### PR TITLE
Document how to reconnect the USB for 32u4 based boards

### PR DESCRIPTION
**Pull request by jeraymond**
For 32u4 based boards the SleepDog library is [cleaning detaching from the USB](https://github.com/jeraymond/Adafruit_SleepyDog/blob/master/utility/WatchdogAVR.cpp#L66). This change documents how to re-attach the USB connection for both the board and the host.

### CODE REVIEW

**Comment by jay6621** [comment]
For me while playing with SleepyDog Library on my custom Atsamd21 board(based on Zero) I can see sleep duration in Serial monitor in arduino IDE. Sketch is-
`#include <Adafruit_SleepyDog.h>

void setup() {
  SerialUSB.begin(115200);
  while (!SerialUSB) ; // wait for Arduino Serial Monitor (native USB boards)
  SerialUSB.println("Adafruit Watchdog Library Sleep Demo!");
  SerialUSB.println();
}

void loop() {
  SerialUSB.println("Going to sleep in one second...");
  delay(1000);

  USBDevice.detach();
  int sleepMS = Watchdog.sleep();

  USBDevice.attach();
  delay(5000);
  while(!SerialUSB);

  SerialUSB.print("I'm awake now!  I slept for ");
  SerialUSB.print(sleepMS, DEC);
  SerialUSB.println(" milliseconds.");
  SerialUSB.println();
  SerialUSB.flush();
  delay(5000);
}`
we need to restart serial monitor in arduino IDE on every attach sound.

**Comment by jay6621** [comment]
For me while playing with SleepyDog Library on my custom Atsamd21 board(based on Zero) I can see sleep duration in Serial monitor in arduino IDE. Sketch is-
```
#include <Adafruit_SleepyDog.h>

void setup() {
  SerialUSB.begin(115200);
  while (!SerialUSB) ; // wait for Arduino Serial Monitor (native USB boards)
  SerialUSB.println("Adafruit Watchdog Library Sleep Demo!");
  SerialUSB.println();
}

void loop() {
  SerialUSB.println("Going to sleep in one second...");
  delay(1000);

  USBDevice.detach();
  int sleepMS = Watchdog.sleep();

  USBDevice.attach();
  delay(5000);
  while(!SerialUSB);

  SerialUSB.print("I'm awake now!  I slept for ");
  SerialUSB.print(sleepMS, DEC);
  SerialUSB.println(" milliseconds.");
  SerialUSB.println();
  SerialUSB.flush();
  delay(5000);
}
```
we need to restart serial monitor in arduino IDE on every attach sound.
